framework/dora_api/infrastructure/middleware.py

```diff
@@ -38,18 +39,42 @@ async def deserialise_web_request():
     _RequestEndpoint = request.endpoint.split(".")[-1]
     if _RequestEndpoint in REQUEST_BODYS_BY_ENDPOINT:
         _RequestData: dict = request.get_json()
+        _DeserialisedRequestData : dict = dict()
 
         for _AttributeName, _AttributeType in get_type_hints(REQUEST_BODYS_BY_ENDPOINT[_RequestEndpoint]).items():
-            _Data = _RequestData[_AttributeName]
+            _Data = _RequestData.get(_AttributeName)
+
+            _AttributeTypeOrigin = get_origin(_AttributeType)
+            if _AttributeTypeOrigin is not None and issubclass(_AttributeTypeOrigin, AttributeChangeTracker):
+                __deserialise_attribute_change_tracker(_AttributeName, _RequestData, _DeserialisedRequestData)
+                continue
 
             if _AttributeType is bytes:
                 _Data = b64decode(_Data)
 
-            _RequestData[_AttributeName] = _AttributeType(_Data) if _Data else None
+            _DeserialisedRequestData[_AttributeName] = _AttributeType(_Data) if _DeserialisedRequestData else None
```

Comment:
Hmm this if doesn't make sense, remnant of old code?

Reply:
I think it's meant to be data still

Reply:
yeah, my bad